pkg/pool-weighted/contracts/LBPool.sol

```diff
@@ -0,0 +1,294 @@
+        address trustedRouter
+    ) WeightedPool(params, vault) Ownable(owner) {
+        // WeightedPool validates `numTokens == normalizedWeights.length`, and ensures valid weights.
+        // Here we additionally enforce that LBPs must be two-token pools.
+        InputHelpers.ensureInputLengthMatch(_NUM_TOKENS, params.numTokens);
+
+        // Set the trusted router (passed down from the factory).
+        _TRUSTED_ROUTER = trustedRouter;
+
+        // solhint-disable-next-line not-rely-on-time
+        uint32 currentTime = block.timestamp.toUint32();
+        _startGradualWeightChange(currentTime, currentTime, params.normalizedWeights, params.normalizedWeights);
+        _setSwapEnabled(swapEnabledOnStart);
+    }
+
+    /**
+     * @notice Return start time, end time, and endWeights as an array.
+     * @dev Current weights should be retrieved via `getNormalizedWeights()`.
+     * @return startTime The starting timestamp of any ongoing weight change
+     * @return endTime The ending timestamp of any ongoing weight change
+     * @return endWeights The "destination" weights, sorted in token registration order
+     */
+    function getGradualWeightUpdateParams()
+        external
+        view
+        returns (uint256 startTime, uint256 endTime, uint256[] memory endWeights)
+    {
+        PoolState memory poolState = _poolState;
+
+        startTime = poolState.startTime;
+        endTime = poolState.endTime;
+
+        endWeights = new uint256[](_NUM_TOKENS);
+        endWeights[0] = poolState.endWeight0;
+        endWeights[1] = FixedPoint.ONE - poolState.endWeight0;
+    }
+
+    /**
+     * @notice Indicate whether swaps are enabled or not for the given pool.
+     * @return swapEnabled True if trading is enabled
+     */
+    function getSwapEnabled() external view returns (bool) {
+        return _getPoolSwapEnabled();
+    }
+
+    /*******************************************************************************
+                                Permissioned Functions
+    *******************************************************************************/
+
+    /**
+     * @notice Enable/disable trading.
+     * @dev This is a permissioned function that can only be called by the owner.
+     * @param swapEnabled True if trading should be enabled
+     */
+    function setSwapEnabled(bool swapEnabled) external onlyOwner {
+        _setSwapEnabled(swapEnabled);
+    }
+
+    /**
+     * @notice Start a gradual weight change. Weights will change smoothly from current values to `endWeights`.
+     * @dev This is a permissioned function that can only be called by the owner.
+     * If the `startTime` is in the past, the weight change will begin immediately.
+     *
+     * @param startTime The timestamp when the weight change will start
+     * @param endTime  The timestamp when the weights will reach their final values
+     * @param endWeights The final values of the weights
+     */
+    function updateWeightsGradually(
```

Comment:
Super minor, but since this has a `startTime`, you'd think it should be named something like `scheduleGradualWeightsUpdate`

Reply:
This is using the legacy terminology, all the way back to v1. If it's not important to maintain that, it could be renamed.

Reply:
Looks like @EndymionJkb beat me to it. I personally don't mind if we want to modernize the function name; I was just trying to maintain the [v2 function names](https://github.com/balancer/balancer-v2-monorepo/blob/master/pkg/pool-weighted/contracts/lbp/LiquidityBootstrappingPoolSettings.sol#L206-L219) for easiest interfacing with existing integrators

Reply:
resolving and adding as a comment to https://github.com/balancer/balancer-v3-monorepo/pull/971